### PR TITLE
docs: replacement of method type_as in docs to Tensor.to

### LINKED ISSUE #14554: Replace mentions of `.type_as()` in our docs
## 📚 Documentation

As a follow up to #2585, we should consider removing mentions of the `.type_as()` syntax in our docs and replace it with best practices for device placement and type conversion. 

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

cc @borda @rohitgr7 @Felonious-Spellfire

### PR DESCRIPTION
**Pull request by amrutha1098**
## What does this PR do?

Replaces the .type_as with Tensor.to

Fixes #14554

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @awaelchli @nsarang

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: docs/source-pytorch/accelerators/accelerator_prepare.rst [hunk 1]
```diff
@@ -48,7 +48,7 @@ This will make your code scale to any arbitrary number of GPUs or TPUs with Ligh
     # with lightning
     def forward(self, x):
         z = torch.Tensor(2, 3)
-        z = z.type_as(x)
+        z = z.Tensor.to(x)
```
Comment (awaelchli) [inline docs/source-pytorch/accelerators/accelerator_prepare.rst:51]:
```suggestion
        z = z.to(x)
```

File: docs/source-pytorch/common/lightning_module.rst [hunk 2]
```diff
@@ -55,7 +55,7 @@ Notice a few things.
 
         # or to init a new tensor
         new_x = torch.Tensor(2, 3)
-        new_x = new_x.type_as(x)
+        new_x = new_x.Tensor.to(x)
```
Comment (awaelchli) [inline docs/source-pytorch/common/lightning_module.rst:58]:
```suggestion
        new_x = new_x.to(x)
```

**Review by awaelchli — APPROVED** [review]
